lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
```

Comment:
> Yes, it won't be as fast as just using  instanceof , but it will still be faster :)

Yea, I figured that the non-array case would get extra slow from two checks. I wonder if throwing a `typeof values === 'object'` somewhere will help in that case. Perhaps a `typeof values === 'object' && (values instanceof Array || Array.isArray(values))`?